bin/check_fasta_insdc.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python
+
+import argparse
+import gzip
+import hashlib
+
+from itertools import groupby
+
+import requests
+from ebi_eva_common_pyutils.logger import logging_config
+
+import yaml
+from requests import HTTPError
+from retry import retry
+
+REFGET_SERVER = 'https://www.ebi.ac.uk/ena/cram'
+
+logger = logging_config.get_logger(__name__)
+
+
+def open_gzip_if_required(input_file):
+    if input_file.endswith('.gz'):
+        return gzip.open(input_file, 'rt')
+    else:
+        return open(input_file, 'r')
+
+
+def write_result_yaml(output_yaml, results):
+    with open(output_yaml, 'w') as open_yaml:
+        yaml.safe_dump(data=results, stream=open_yaml)
+
+
+def refget_md5_digest(sequence):
+    return hashlib.md5(sequence.upper().encode('utf-8')).hexdigest()
+
+
+def fasta_iter(input_fasta):
+    """
+    Given a fasta file. yield tuples of header, sequence
+    """
+    "first open the file outside "
+    fin = open(input_fasta, 'r')
+
+    # ditch the boolean (x[0]) and just keep the header or sequence since
+    # we know they alternate.
+
+    faiter = (x[1] for x in groupby(fin, lambda line: line[0] == ">"))
+
+    for header in faiter:
+        # drop the ">"
+        headerStr = header.__next__()[1:].strip()
+
+        # join all sequence lines to one.
+        seq = "".join(s.strip() for s in faiter.__next__())
+        yield (headerStr, seq)
```

Comment:
This is clever, though I must confess I read it through a number of times in utter confusion before checking the docs to see that `groupby` only groups consecutive elements

Reply:
Yes it was clever indeed: https://www.biostars.org/p/710/#120760